lnrpc/walletrpc/psbt.go

```diff
@@ -36,8 +38,7 @@ func verifyInputsUnspent(inputs []*wire.TxIn, utxos []*lnwallet.Utxo) error {
 		}
 
 		if !found {
-			return fmt.Errorf("input %d not found in list of non-"+
-				"locked UTXO", idx)
+			return status.Error(codes.NotFound, ErrUnspentInputNotFound(idx).Error())
```

Comment:
I don't think this will return a structured error to the client and they still need to parse the text message to find out the index?

Reply:
I guess it depends what you mean by a structured error. On the client side, they can introspect the error and extract the error code used and act upon then, doing string parsing if they need any additional context. 

Unless you mean return a _response_ that has fields to enumerate the different types of errors?

Reply:
Did some digging in the docs, and looks like it's possible for us to actually get the _best_ of both worlds here: https://pkg.go.dev/google.golang.org/grpc/internal/status?utm_source=godoc#Status.WithDetails 

This API lets you make a status code to return using `status.Error`, but then also attach an _arbitrary_ proto message that can be used to let the client optionally get more structured information if it needs to. This blog post has a good overview of how things would work end to end: https://jbrandhorst.com/post/grpc-errors/

Reply:
Yes, `WithDetails` is what I used before indeed for that. I think that can come in useful in various places, because not everything maps cleanly to a grpc code. And in this case, the index needs to be stored somewhere. If an utxo is locked, it is likely that a client needs info about which one it is exactly to do another funding attempt.

Reply:
If think the requirement here is that systems should be able to respond appropriately to a specific utxo being unavailable. Just a generic error code wouldn't suffice then.

Reply:
I didn't know about the `WithDetails`, could be useful indeed. Though the fact that it needs to be a proto message would mean we'd need to add new messages to our main proto file just for structured errors? Might get bloated quite quickly.

Also, I wonder how such an error message (with details) would look on the command line, if the error is just printed?

Reply:
There may be many unique error cases, but all the 'unexpected' ones - the internal errors - don't need to be returned in a structured way I'd say. I am not sure if that many structured errors remain.